application/models/Question.php

```diff
@@ -126,7 +126,13 @@ public function relations()
             'parent' => array(self::HAS_ONE, 'Question', array("qid" => "parent_qid")),
             'questionattributes' => array(self::HAS_MANY, 'QuestionAttribute', 'qid'),
             'questionl10ns' => array(self::HAS_MANY, 'QuestionL10n', 'qid', 'together' => true),
-            'subquestions' => array(self::HAS_MANY, 'Question', array('parent_qid' => 'qid'), 'order' => App()->getDb()->quoteColumnName('subquestions.question_order') . ' ASC'),
+            'subquestions' => array(
```

Comment:
Is that for lazy loading?

Reply:
together => false provides active record attempting to handle the relation with a join rather than separate query. Without this set, the query fails due to running out of memory.

Reply:
Hm, how much memory, and where?

Reply:
So probably 1-2 GB? 

Is this a fix for the REST API or a bug fix for some other scenario?

Reply:
Maybe? :) Using JOIN creates fewer calls to the database, I'm assuming? So it's a trade-off, depending on the scenario.

Reply:
One-to-many relationship is not a good candidate for a join when you have multiple one-to-many relationships in a single query. I think that's generally true.

Reply:
Still, performance consequences on cloud is not 100% clear to me. Maybe ask Carsten too just in case.

Reply:
I think you don't understand exactly what this change does. I don't believe there is any reason for concern about performance. I have sent message to Carsten to get his feedback regardless.

Reply:
Question is how many more queries will happen during createFieldmap for example, when the JOIN is disabled. But yeah, perhaps I'm wrong. :)

Reply:
This only affects upfront loading using find() and findAll() via with('subquestions') if you use $question->subquestions it is a separate query regardless. Looking in the code there are only a few places using with('subquestions').

When using find() and findAll() via with('subquestions') it is a single additional query for all related subquestions. If you are loading 50 questions with sub questions there will be one query for the questions and one query for all of the subquestions (not for query for subquestions of each question / + 50 queries).

Reply:
I can find any code in-use which is pre-loading subquestions. 

There is:
`application/controllers/SurveyAdministrationController.php:3249:        $oResult = Question::model()->with('subquestions')->findAll(`

From SurveyAdministrationController::tabPanelIntegration() but this method doesn't appear to be called from anywhere.

Since no existing code relies on pre-loading subquestions, my change does not impact any existing functionality.